src/app/components/authorized/search-bar/search-bar.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, Input, OnChanges, OnInit, Output, SimpleChanges } from '@angular/core';
+import { Router } from '@angular/router';
+import { GatewayService } from 'src/app/services/gateway.service';
+import { SearchableUser } from 'src/types/user';
+
+@Component({
+    selector: 'app-search-bar',
+    templateUrl: './search-bar.component.html',
+    styleUrls: ['./search-bar.component.scss'],
+})
+export class SearchBarComponent implements OnInit {
+    users: SearchableUser[] = [];
+    searchText = '';
+    searchResults: SearchableUser[] = [];
+
+    constructor(private _gatewayService: GatewayService, private _router: Router) {}
+
+    ngOnInit(): void {
+        this._gatewayService.getAllUsers().subscribe((users) => {
+            this.users = users;
+            this.searchResults = users;
+        });
+    }
+
+    onTyping(): void {
+        this.searchResults = this.users.filter((user) =>
```

Comment:
This is totally fine with the current state of things, so you don't need to update this. I'd just like to point out that this changes on every single keystroke, meaning that if you'd have ~10k users (or some bigger number in general), this would run for every single keystroke. Your user may have a typo etc., so this is not really the best way. Nevertheless, for the scale of this project, it is sufficient.